src/lib/data/index.ts

```diff
@@ -268,3 +270,189 @@ export function useSettings(): {
 
     return { settings, updateSettings, loadSettings, handlers };
 }
+
+export interface AlertList {
+    [date: string]: string[];
+}
+
+export const defaultAlertList = {};
+
+export function useAlertList(): {
+    alertList: AlertList;
+    addAlert: (date: string, id: string) => void;
+    removeAlert: (date: string, id: string) => void;
+    removeDateKey: (date: string) => void;
+    loadAlertList: () => Promise<void>;
+    pollAlertList: (eventsQuery: QueryResult<EventListQuery, EventListQueryVariables>, interval?: number) => void;
+} {
+    const storage = useStorage();
+    const [alertList, setAlertList] = useState<AlertList>(defaultAlertList);
+
+    // The time of the event(s) has come (most likely),
+    // so we are being told to remove the dynamic date (as an obj property), as well as from the dateKeys array
+    const removeDateKey = useCallback(
+        (date: string) => {
+            setAlertList((state) => {
+                const cleanState: AlertList = { ...state };
```

Comment:
Since you're defining `alertList` as a dependency for this callback, should you be using that instead of `state` here?